langchain4j-core/src/test/java/dev/langchain4j/web/search/WebSearchToolIT.java

```diff
@@ -0,0 +1,25 @@
+package dev.langchain4j.web.search;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class WebSearchToolIT extends WebSearchIT{
+
+    @Test
+    void should_return_pretty_result_as_a_tool(){
```

Comment:
I would make it a real IT with `ChatLanguageModel`, to make sure the tool is usable with the LLM

Reply:
yeah I agree, I've add a real `ChatLanguageModel` and I force that LLM recognize it as a tool and convert from a  `ToolSpecification` to a `ToolExecutionRequest`.

It'd looks better if we can use `AiServices` to execute the tool but due this IT is in core, I believe will generate us cycle calling issue.